src/lib/version.ts

```diff
@@ -0,0 +1,132 @@
+import debugFactory from 'debug';
+import semver from 'semver';
+
+const debug = debugFactory('clark:lib:version');
+
+/**
+ * Selcts the greater of two semver ranges combined with their most permissive
+ * range modifier
+ * @param left
```

Comment:
Don't need to worry about params with typescript. Can remove throughout this PR.

Reply:
vscode added them automatically